### ISSUE TITLE
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address

### ISSUE THREAD

**Issue by alittlesir** [python/cpython#102167]
# Bug report

when using gdb to debug python programs and `py-bt` does not work. and ouput
```
[root@localhost aaa]# gdb python3.7dm core.920189 
GNU gdb (GDB) EulerOS 9.2-1.h4.eulerosv2r10
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "aarch64-openEuler-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python3.7dm...
Reading symbols from /usr/lib/debug//usr/bin/python3.7dm-3.7.9-9.h29.eulerosv2r10.aarch64.debug...

warning: core file may not match specified executable file.
[New LWP 920189]

warning: File "/usr/lib64/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".
To enable execution of this file add
	add-auto-load-safe-path /usr/lib64/libthread_db-1.0.so
line to your configuration file "/root/.gdbinit".
To completely disable this security protection add
	set auto-load safe-path /
line to your configuration file "/root/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
	info "(gdb)Auto-loading safe path"

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.

warning: File "/usr/lib64/libthread_db-1.0.so" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".

warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
Core was generated by `python3'.
#0  0x0000ffff90fce228 in select () from /usr/lib64/libc.so.6
(gdb) py-bt
Traceback (most recent call first):
  <unknown at remote 0xffff840653c0>
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x6200000081: 
Error occurred in Python: Cannot access memory at address 0x6200000081
(gdb)

```
python program, only a simple file: test.py
```
import time


def do(x):
    time.sleep(10)


def main():
    for x in range(10000):
        do(x)


if __name__ == '__main__':
    main()
```

`py-bt` can't work, what is wrong? because it is compile with `-Og` ?

# Your environment

- CPython versions tested on: python3.7.7
- Operating system and architecture: openeuler arm64

**Comment by terryjreedy** [comment]
What is py-bt and what does it have to do with the simple but long running test.py?  Is it really necessary to test for over a day?

3.9 and earlier only get security fixes; please test with 3.10 or later.

**Comment by alittlesir** [comment]
oh, i am sorry, I'm just using this example(test.py) to make a demonstration. The real program  is `firewalld`.
1、systemctl start firewalld
2、ps aux | grep firewalld （firewalld process）
[root@localhost ~]# ps aux | grep firewalld
root        1174  0.0  1.1 329500 39608 ?        Ssl  2月22   0:00 /usr/bin/python3 /usr/sbin/firewalld --nofork --nopid

3、gcore  1174  (Core files are generated，core.1174)
4、yum install python3-debuginfo
5、gdb /usr/bin/python3
6、source /root/libpython.py
7、py-bt

and ouput 
```
(gdb) py-bt
Traceback (most recent call first):
  <unknown at remote 0xffff840653c0>
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x6200000081: 
Error occurred in Python: Cannot access memory at address 0x6200000081
```

**Comment by alittlesir** [comment]
py-bt，py-list, py-locals doesn't all work

```
(gdb) py-bt
Traceback (most recent call first):
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x6200000081: 
Error occurred in Python: Cannot access memory at address 0x6200000081
(gdb) py-list
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x6200000081: 
Error occurred in Python: Cannot access memory at address 0x6200000081
(gdb) py-locals
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x6200000081: 
Error occurred in Python: Cannot access memory at address 0x6200000081

```

**Comment by iritkatriel** [comment]
> 3.9 and earlier only get security fixes; please test with 3.10 or later. If you cannot, please close.

**Comment by alittlesir** [comment]
yehh, maybe you are right, which commit was this fixed? I need a patch on 3.7

**Comment by iritkatriel** [comment]
We don’t know what it is. If you want us too look at bug reports you need to upgrade to a suppprted version.
